autogen/agentchat/conversable_agent.py

```diff
@@ -131,7 +135,11 @@ def __init__(
 
         self._name = name
         # a dictionary of conversations, default value is list
-        self._oai_messages = defaultdict(list)
+        if messages is None:
+            self._oai_messages = defaultdict(list)
+        else:
+            self._oai_messages = messages
```

Comment:
Make deepcopy of the messages in case these messages are shared across multiple agents.